src/bundle/eZ/RichText/Renderer.php

```diff
@@ -216,6 +236,23 @@ public function renderLocationEmbed($locationId, $viewType, array $parameters, $
 
                 return null;
             }
+        } catch (AuthenticationCredentialsNotFoundException $e) {
+            // There is no firewall for 404 page, and we're unable to use "isGranted".
```

Comment:
it looks like this try/catch construction has grown a lot, it could be good to split it a bit into private functions